query-node/mappings/src/content/video.ts

```diff
@@ -0,0 +1,355 @@
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  // read event data
+  const {videoCategoryId} = new Content.VideoCategoryDeletedEvent(event).data
+
+  // load video category
+  const videoCategory = await db.get(VideoCategory, { where: { id: videoCategoryId } })
+
+  // ensure video category exists
+  if (!videoCategory) {
+    return inconsistentState('Non-existing video category deletion requested', videoCategoryId)
+  }
+
+  // remove video category
+  await db.remove<VideoCategory>(videoCategory)
+
+  // emit log event
+  logger.info('Video category has been deleted', {id: videoCategoryId})
+}
+
+/////////////////// Video //////////////////////////////////////////////////////
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_VideoCreated(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  // read event data
+  const {
+    channelId,
+    videoId,
+    videoCreationParameters,
+    contentActor,
+  } = new Content.VideoCreatedEvent(event).data
+
+  // read metadata
+  const protobufContent = await readProtobuf(
+    new Video(),
+    videoCreationParameters.meta,
+    videoCreationParameters.assets,
+    db,
+    event.blockNumber,
+    contentActor,
+  )
+
+  // create new video
+  const video = new Video({
+    id: videoId,
+    isCensored: false,
+    channel: channelId,
+    happenedIn: event.blockNumber,
+    ...Object(protobufContent)
+  })
+
+  // save video
+  await db.save<Video>(video)
+
+  // emit log event
+  logger.info('Video has been created', {id: videoId})
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_VideoUpdated(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  // read event data
+  const {
+    videoId,
+    videoUpdateParameters,
+    contentActor,
+  } = new Content.VideoUpdatedEvent(event).data
+
+  // load video
+  const video = await db.get(Video, { where: { id: videoId } })
+
+  // ensure video exists
+  if (!video) {
+    return inconsistentState('Non-existing video update requested', videoId)
+  }
+
+  // prepare changed metadata
+  const newMetadata = videoUpdateParameters.new_meta.isSome && videoUpdateParameters.new_meta.unwrapOr(null)
+
+  // update metadata if it was changed
+  if (newMetadata) {
+    const protobufContent = await readProtobuf(
+      new Video(),
+      newMetadata,
+      videoUpdateParameters.assets.unwrapOr([]),
+      db,
+      event.blockNumber,
+      contentActor,
+    )
+
+    // remember original license
+    const originalLicense = video.license
+
+    // update all fields read from protobuf
+    for (let [key, value] of Object(protobufContent).entries()) {
+      video[key] = value
+    }
+
+    // license has changed - delete old license
+    if (originalLicense && video.license != originalLicense) {
+      await db.remove<License>(originalLicense)
+    }
+  }
+
+  // TODO: handle situation when only assets changed
```

Comment:
When video is updated this event parameter is provided:
```
export interface VideoUpdateParameters extends Struct {
  readonly assets: Option<Vec<NewAsset>>;
  readonly new_meta: Option<Bytes>;
}
```

This seems like only assets (without meta) can be updated. But what would that mean? Metadata usually contains asset indices for assets that should be associated with, let's say cover photo, etc.

Reply:
Seems, that there is no way for runtime to ensure the correctness between metadata and links to the respective assets. 

I guess, that means, that metadata and it's indices would remain the same as you have now, but links to the respective assets would be changed. 
Same applies to `Channels`.

Reply:
Ok. So if I understand this correctly, there is no need to handle this situation in mappings? (assets are set while metadata are not)